### PR TITLE
chore: use Lucide icons for Buttons and refractor code

### PR DESCRIPTION
**Pull request by VIKTORVAV99**
## Issue

We used old non Lucide icons here and had some duplicate code.

Part of: AVO-439

## Description

Switches to Lucide icons and deduplicates the Commercial API Button code into its own component + stories for it.

This reduces the total app size from `4071.77 KiB` to `4071.61 KiB`.

### Double check

- [x] I have run `pnpx prettier@2 --write .` and `poetry run format` in the top level directory to format my changes.

### CODE REVIEW

**Review by silkeholmebonnen — COMMENTED** [review]
Looks good

File: web/src/features/panels/zone/ZoneDetails.tsx [hunk 1]
```diff
@@ -120,13 +109,7 @@ export default function ZoneDetails(): JSX.Element {
...
           <Attribution zoneId={zoneId} />
           {isMobile ? (
-            <Button
-              backgroundClasses="mt-3"
-              icon={<MdOutlineCloudDownload size={20} />}
-              href="https://electricitymaps.com/?utm_source=app.electricitymaps.com&utm_medium=referral&utm_campaign=country_panel"
-            >
-              {t('header.get-data')}
-            </Button>
+            <CommercialApiButton backgroundClasses="mt-3" />
```
[the remark this answers is not in the document]
Comment (silkeholmebonnen) [reply web/src/features/panels/zone/ZoneDetails.tsx:112]: This comment is for both of the CommercialApiButtons

[the remark this answers is not in the document]
Comment (VIKTORVAV99) [reply web/src/features/panels/zone/ZoneDetails.tsx:112]: Should be the same size as it was before now 🙂

**Review by silkeholmebonnen — APPROVED** [review]
Nice work!

**Pull request merged**